### ISSUE TITLE
Clarify what 'property value is set directly to the attribute' means

### ISSUE THREAD

**Issue by kenchris** [Polymer/lit-element#166]
> If no toAttribute function is provided and reflect is set to true, the property value is set directly to the attribute.

I assume this will call the `toString()` by default.

Testing with 
`$0.setAttribute("hi", new Object)` yields `[object Object]`
and 
`$0.setAttribute("hi", true)` yields `true`

But it might also throw (ie. if I make a mistake like below):

![image](https://user-images.githubusercontent.com/1101339/44906304-cfad6a80-ad14-11e8-9eb5-dc62ae1de8db.png)

`$0.setAttribute("hi", new Object({ toString: () => "hi" }))` becomes `hi` (string)

But apart from that, you do have special handling of Boolean, which should probably be mentioned here as well:

```
    const toAttribute = options.type === Boolean ? toBooleanAttribute :
        (options.type && (options.type as AttributeSerializer).toAttribute || String);
```

**Comment by TimPietrusky** [comment]
@kenchris I still don't understand what I have to do in order to pass an Object as an attribute into a component. But I guess just defining the `type: Object` is not enough, because that is giving me a `[object Object]`. 

Version: 0.6.0-dev.6

**Comment by web-padawan** [comment]
@TimPietrusky you should be able to implement the Object type yourself, based in the following parts of Polymer classic library:

https://github.com/Polymer/polymer/blob/2a49743362184062025984446e1b8311eddf9235/lib/mixins/property-accessors.js#L199

https://github.com/Polymer/polymer/blob/2a49743362184062025984446e1b8311eddf9235/lib/mixins/property-accessors.js#L239

e.g. something like 
```js
type: {
  fromAttribute: value => {
    let outValue;
    try {
      outValue = JSON.parse(/** @type {string} */(value));
    } catch(x) {
      // allow non-JSON literals like Strings and Numbers
      outValue = value;
    }
  }
}
```
However, you please keep in mind that lit-element does not provide dirty-checking for you.

**Comment by aadamsx** [comment]
> lit-element does not provide dirty-checking for you

What is Dirty-checking?

**Comment by TimPietrusky** [comment]
@web-padawan I found the solution for my problem while checking the commit in the pwa-starter-kit project (which I used as the basis for my project). They are using lit-element@0.5.2, but migrated to 0.6.0-dev.5 a couple of weeks ago. And in their demo project they have the exact use case as I described above. 

In [this commit](https://github.com/Polymer/pwa-starter-kit/blob/6401a67406d98918af44c1d14b5b6083040c072b/src/components/my-view3.js#L72) you find the answer:

```
 <shop-cart .products="${_products}" .cart="${_cart}"></shop-cart>
```

They put a `.` as a prefix to the attribute (which is of `type: Object`) and then passing an Object as the value. 

I couldn't find any reference why this is the case, but it's fixing my problem.
